### PR TITLE
Fix NDArrayIter.__init__()

### PR DESCRIPTION
**Pull request by Vesnica**
When last_batch_handle == 'discard', self.data is a list, so self.data[str] will fail, should use self.data_list[0/1]

### CODE REVIEW

**Comment by piiswrong** [comment]
This fix seems too hacky. The discard part is outdated and should be changed to using dictionaries properly.

File: python/mxnet/io.py [hunk 1]
```diff
@@ -284,9 +284,9 @@ def __init__(self, data, label=None, batch_size=1, shuffle=False, last_batch_han
         if last_batch_handle == 'discard':
             new_n = self.data_list[0].shape[0] - self.data_list[0].shape[0] % batch_size
             for k, _ in self.data:
-                self.data[k] = self.data[k][:new_n]
+                self.data[k] = self.data_list[0][:new_n]
             for k, _ in self.label:
-                self.label[k] = self.label[k][:new_n]
+                self.label[k] = self.data_list[1][:new_n]
```
Comment (winstywang) [inline python/mxnet/io.py]: Should it be label_list or etc?

**Comment by Vesnica** [comment]
@piiswrong

**Comment by winstywang** [comment]
Sorry, need to update again to merge in

**Comment by Vesnica** [comment]
@winstywang All ready

**Pull request merged**